src/components/view-pane/related-views-button.tsx

```diff
@@ -18,25 +22,38 @@ export class RelatedViewsButtonBase extends React.PureComponent<RelatedViewsButt
   }
 
   public render() {
+    const {config, relatedViews} = this.props;
+    const {hideRelatedViews} = config;
+    const hideCondition = (relatedViews.isHidden === undefined && hideRelatedViews) ||
+      (relatedViews.isHidden !== undefined && relatedViews.isHidden);
     return (
       <div styleName="right">
         <a onClick={this.onHideClick}>
-          {this.props.relatedViews.isHidden ? 'Show' : 'Hide'}
+          {hideCondition ? "Show" : "Hide"}
           &nbsp;&nbsp;
-          {!this.props.relatedViews.isHidden ?
-            <i className="fa fa-toggle-down" /> :
-            <i className="fa fa-toggle-up" />
-          }
+          {hideCondition ? <i className="fa fa-toggle-up"/> : <i className="fa fa-toggle-down"/>}
         </a>
       </div>
     );
   }
 
   private onHideClick() {
+    const {relatedViews, config} = this.props;
     this.props.handleAction({
-      type: RELATED_VIEWS_HIDE_TOGGLE
+      type: RELATED_VIEWS_HIDE_TOGGLE,
+      payload: {
+        hideRelatedViews: relatedViews.isHidden === undefined ? config.hideRelatedViews : relatedViews.isHidden
+      }
     });
   }
 }
 
-export const RelatedViewsButton = (CSSModules(RelatedViewsButtonBase, styles));
+export const RelatedViewsButton = connect(
```

Comment:
This "connect" is unnecessary.  (Better change the properties to have the "hide" state like commented above, that's it.)